src/scripts/components/main/data-viewer/data-viewer.tsx

```diff
@@ -87,59 +95,79 @@ const Globes: FunctionComponent<Props> = ({backgroundColor, markers = []}) => {
     setCurrentView(globalGlobeView);
   }, [globalGlobeView]);
 
-  return (
-    <div className={styles.globes}>
-      {mainLayerDetails && (
-        <LayerLegend
-          id={mainLayerDetails.id}
-          values={[mainLayerDetails.maxValue, mainLayerDetails.minValue]}
-          unit={mainLayerDetails.units}
-        />
-      )}
-      {compareLayerDetails && (
-        <LayerLegend
-          id={compareLayerDetails.id}
-          values={[compareLayerDetails.maxValue, compareLayerDetails.minValue]}
-          unit={compareLayerDetails.units}
-          isCompare={true}
-        />
-      )}
+  const showGlobeNavigation =
+    (!mainLayerDetails && !compareLayerDetails) ||
+    [mainLayerDetails, compareLayerDetails].some(
+      layer => layer && layer.type !== LayerType.Gallery
+    );
+
+  const getDataWidget = ({
+    imageLayer,
+    layerDetails,
+    active,
+    action
+  }: {
+    imageLayer: GlobeImageLayerData | null;
+    layerDetails: Layer | null;
+    active: boolean;
+    action: () => void;
+  }) => {
+    if (imageLayer?.type === LayerType.Gallery) {
+      return <Gallery imageLayer={imageLayer} />;
+    }
+
+    return (
       <Globe
         markers={markers}
         backgroundColor={backgroundColor}
-        active={isMainActive}
+        active={active}
         view={currentView}
         projectionState={projectionState}
-        imageLayer={mainImageLayer}
-        basemap={mainLayerDetails?.basemap || null}
+        imageLayer={imageLayer}
+        basemap={layerDetails?.basemap || null}
         spinning={globeSpinning}
         flyTo={flyTo}
-        onMouseEnter={() => setIsMainActive(true)}
-        onTouchStart={() => setIsMainActive(true)}
+        onMouseEnter={action}
+        onTouchStart={action}
         onChange={onChangeHandler}
         onMoveEnd={onMoveEndHandler}
         onMouseDown={onMouseDownHandler}
       />
+    );
+  };
+
+  return (
+    <div className={styles.dataViewer}>
+      {[mainLayerDetails, compareLayerDetails]
+        .filter((layer): layer is Layer => Boolean(layer))
```

Comment:
is this the same as `.filter((layer: Layer) => Boolean(layer))`?

Reply:
No it's not. I was fighting typescript on this one. I never understand why typescript (and also flow) can not infer this kind of stuff. I clearly filter out all the `null` layers and then it complains in the VERY NEXT LINE that it could be null :D

<img width="1140" alt="Bildschirmfoto 2020-08-31 um 11 56 28" src="https://user-images.githubusercontent.com/2902795/91708143-2eff1900-eb81-11ea-9d80-dc79b40bc1fb.png">

Reply:
I didn't know this pattern before. Its called type predicates or custom type guards. https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
ah ok, yes strange that this does not work out of the box